src/functional/iterator/embedded.py

```diff
@@ -586,27 +603,31 @@ def shape(self):
         return self.array().shape
 
 
+def _is_field_axis(axis: Axis) -> TypeGuard[FieldAxis]:
+    return isinstance(axis, (Dimension, Offset))  # TODO should be `isinstance(axis, FieldAxis)`
```

Comment:
What's the correct pattern here, I want to use the alias FieldAxis in isinstance check, but mypy complains.

Reply:
https://github.com/python/mypy/issues/11673